feat(ssh_tunnel): Rename allow_ssh_tunneling and change the default value to False

Antonio-RiveroMartnez opened the pull request:
### SUMMARY
- Rename the `allow_ssh_tunneling` flag so it's now `disable_ssh_tunneling`, always false except for those DB engines we know it's not supported (Amazon Athena, GSheets and BigQuery so far)
- Update tests

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TESTING INSTRUCTIONS
<!--- Required! What steps can be taken to manually verify the changes? -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Antonio-RiveroMartnez committed:
SSH Tunnel UI:

- Change the default value of allow_ssh_tunneling flag so it's always true except for those DB engines we know it's not supported

eschutho commented on superset/db_engine_specs/base.py:
```diff
@@ -193,7 +193,7 @@ class BaseEngineSpec:  # pylint: disable=too-many-public-methods
     drivers: Dict[str, str] = {}
     default_driver: Optional[str] = None
-    allow_ssh_tunneling = False
+    allow_ssh_tunneling = True
```
This is kind of a nit, but I think your first inclination to make the default value `False` was a good one and follows convention. Can we instead flip the behavior, so something like `disable_ssh_tunneling = False` (or some name that is shorter. :)

Antonio-RiveroMartnez replied:
Oh! Ok, yeah, makes sense, let me change it

Antonio-RiveroMartnez committed:
SSH Tunnel UI:

- Change the allow_ssh_tunneling for disable_ssh_tunneling to follow convention as before when the flag was set by default to false

eschutho reviewed (approved):
looks great, thanks @Antonio-RiveroMartnez!
